app-modules/engagement/src/Filament/Resources/EngagementFileResource/RelationManagers/EngagementFilesRelationManager.php

```diff
@@ -115,7 +115,8 @@ public function table(Table $table): Table
                     }),
             ])
             ->headerActions([
-                CreateAction::make(),
+                CreateAction::make()
+                    ->authorize('create', $this->getOwnerRecord()),
```

Comment:
I don't think `$this->getOwnerRecord()` is right here is it is going to be referencing the `Prospect`. Right @danharrin?

Reply:
As above